test/e2e/layering_test.go

```diff
@@ -0,0 +1,98 @@
+package e2e_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	"github.com/openshift/machine-config-operator/test/constants"
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/openshift/machine-config-operator/test/helpers"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestBootInClusterImage(t *testing.T) {
+	cs := framework.NewClientSet("")
+
+	// create a new image stream
+	imageStreamName := "test-boot-in-cluster-image"
+	buildName := imageStreamName
+	imageStreamConfig := &imagev1.ImageStream{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imageStreamName,
+			Namespace: constants.MCONamespace,
+		},
+	}
+	_, err := cs.ImageStreams(constants.MCONamespace).Create(context.TODO(), imageStreamConfig, metav1.CreateOptions{})
+	require.Nil(t, err)
+	defer cs.ImageStreams(constants.MCONamespace).Delete(context.TODO(), imageStreamName, metav1.DeleteOptions{})
+
+	// push a build to the image stream
+	buildConfig := &buildv1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: buildName,
+		},
+		Spec: buildv1.BuildSpec{
+			CommonSpec: buildv1.CommonSpec{
+				Source: buildv1.BuildSource{
+					Type: "Git",
+					Git: &buildv1.GitBuildSource{
+						URI: "https://github.com/cgwalters/fcos-derivation-example",
```

Comment:
Also I am now realizing that this e2e is going to be broken because https://github.com/ostreedev/ostree-rs-ext/pull/196 until I can update the demo image at `quay.io/cgwalters/fcos`, once quay.io is back up.